### PR TITLE
Remove rpc fork and dist autograd fork tests from PyTorch repo

### PR DESCRIPTION
**Pull request by zhaojuanmao**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29827 Remove rpc fork and dist autograd fork tests from PyTorch repo**

There are known issues for "fork tests + OMP" in Pytorch, rpc and dist autograd tests use OMP thread pools, this caused rpc fork and dist autograd fork tests to be flaky. So remove these fork tests from PyTorch repo. rpc spawn and dist autograd spawn tests are still running.

Differential Revision: [D18507384](https://our.internmc.facebook.com/intern/diff/D18507384/)

### CODE REVIEW

**Comment by xush6528** [comment]
Can you do the same to `test_dist_optimizer.py` as well. It also depends on ProcessGroupAgent.

File: test/run_test.py [hunk 1]
```diff
@@ -64,13 +64,11 @@
...
 ]
 
-# skip < 3.3 because mock is added in 3.3 and is used in rpc_fork and rpc_spawn
+# skip < 3.3 because mock is added in 3.3 and is used in rpc_spawn
 # skip python2 for rpc and dist_autograd tests that do not support python2
 if PY33:
     TESTS.extend([
-        'rpc_fork',
         'rpc_spawn',
-        'dist_autograd_fork',
```
Comment (xush6528) [inline test/run_test.py]: Can you do the same to `test_dist_optimizer_fork.py`. It depends on `RpcAgent` as well.

Comment (zhaojuanmao) [reply test/run_test.py]: will do

File: test/run_test.py [hunk 2]
```diff
@@ -64,13 +64,11 @@
     'function_schema',
 ]
 
-# skip < 3.3 because mock is added in 3.3 and is used in rpc_fork and rpc_spawn
+# skip < 3.3 because mock is added in 3.3 and is used in rpc_spawn
 # skip python2 for rpc and dist_autograd tests that do not support python2
 if PY33:
     TESTS.extend([
-        'rpc_fork',
         'rpc_spawn',
```
Comment (mrshenli) [inline test/run_test.py]: since we only have spawn mode on the OSS side, do we still need to explicitly have a "spawn" in the file name?

Comment (zhaojuanmao) [reply test/run_test.py]: we can keep it for now? once we completely get rid of fork, we can merge rpcSpawnTest with RpcTest

**Review by pritamdamania87 — APPROVED** [review]

**Review by mrshenli — APPROVED** [review]
lint and test failures are irrelevant. 

Curious, why dist_optimizer_spawn is not in windows and rocm blacklist?

**Comment by zhaojuanmao** [comment]
I just saw @xush6528 added dist_optim_spawn to blacklist in #29747

**Comment by zhaojuanmao** [comment]
checked "rpc_fork", "dist_autograd_fork" and "dist_optimizer_fork" did not run but "rpc_spawn", "dist_autograd_spawn" and "dist_optimizer_spawn" ran in https://circleci.com/api/v1.1/project/github/pytorch/pytorch/3645637/output/105/0?file=true&allocation-id=5dd30d9f0d6a0b62d1d857f9-0-build%2F10D9680D

**Comment by zhaojuanmao** [comment]
rocm test failure is not relevant 

FAILED (skipped=12, unexpected successes=1)
21:57:37 Traceback (most recent call last):
21:57:37   File "test/run_test.py", line 448, in <module>
21:57:37     main()
21:57:37   File "test/run_test.py", line 441, in main
21:57:37     raise RuntimeError(message)
21:57:37 RuntimeError: test_cuda failed!
21:57:37 + cleanup
21:57:37 + retcode=1
21:57:37 + set +x

**Pull request closed without merging**